backward.hpp

```diff
@@ -3810,11 +3816,15 @@ class SourceFile {
     return paths;
   }
 
-  const std::vector<std::string> &get_paths_from_env_variable() {
+  static std::vector<std::string> &get_mutable_paths_from_env_variable() {
```

Comment:
`static` makes no sense here. What were you trying to do?

Reply:
Adding static here changes almost nothing other than that no object is required to invoke the method (since the body simply returns a static variable of the function, all `.get_paths_from_env_variable()` calls should return the same address no matter what object calls them).

I intended to make it so that no `StackTrace` object was required to invoke this function. For example:
```C++
int main() {
    backward::StackTrace::add_paths_to_env_variable_impl("/home/foo/bar");
    return normal_program();    
}
```
Something like this is what I planned to do for my code. Without static I would have to instantiate `StackTrace` and do something like
```C++
int main() {
    backward::StackTrace s;
    // Changes for all StackTrace objects since the internal variable is static
    s.add_paths_to_env_variable_impl("/home/foo/bar");
    return normal_program();    
}
```

I figured that might be cleaner than setting it per-backtrace, which would require adding this functionality to the constructor and storing the additional paths info in the backtrace object and would split it between the backtrace and the data got in StackTrace from this env variable. My logical reasoning for this as well is that these paths are almost certainly not going to need to be per backtrace but rather set once globally (i.e. via a static variable, which can set by a static method since it would be global and not per-instance); my evidence of this would be that currently it is set by 

If you have a preferred alternative method or desire I can change it however is needed.

Reply:
Ok, I will update the documentation. I'll also add the `volatile` qualifier to the static variable, just in case.

Reply:
The big thing to note is that we have to `const_cast` away the volatility after getting the reference, since vector functions do not support volatile `this` objects. That is, our getter function must return a non-volatile reference.

Reply:
Actually, thinking on it, I'm not entirely sure volatile does anything in this case, since it would only make getting the address a 'volatile operation', but the address is fixed anyway. I can update the PR however though.